packages/nimble-components/src/table/index.ts

```diff
@@ -84,7 +90,36 @@ export class Table<
         this.setTableData(newData);
     }
 
-    public idFieldNameChanged(
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.virtualizer.connectedCallback();
+        this.observeColumns();
+        this.validateColumnIds();
```

Comment:
looks like everywhere we observecolumns we also validate, maybe we should just call validate inside of observe columns? Seems reasonable that the method to setup the observation will validate the result.